2000-06-08  Jakub Jelinek  <jakub@redhat.com>

	* c-typeck.c (digest_init): Don't crash if init is non_lvalue_expr
	of error_mark_node.

--- gcc/c-typeck.c.jj	Wed Jun  7 08:41:04 2000
+++ gcc/c-typeck.c	Thu Jun  8 10:33:52 2000
@@ -4525,7 +4525,11 @@ digest_init (type, init, require_constan
   /* Do not use STRIP_NOPS here.  We do not want an enumerator
      whose value is 0 to count as a null pointer constant.  */
   if (TREE_CODE (init) == NON_LVALUE_EXPR)
-    inside_init = TREE_OPERAND (init, 0);
+    {
+      inside_init = TREE_OPERAND (init, 0);
+      if (inside_init == error_mark_node)
+	return error_mark_node;
+    }
 
   /* Initialization of an array of chars from a string constant
      optionally enclosed in braces.  */
